### ISSUE TITLE
Panic when using SAMPLED_TEXTURE_ARRAY_NON_UNIFORM_INDEXING under DX12

### ISSUE THREAD

**Issue by hasenbanck** [gfx-rs/wgpu-rs#580]
When using texture array's with non-uniform indexing under wgpu master:

```glsl
#version 450
#extension GL_EXT_nonuniform_qualifier : require

layout(location = 0) flat in uint v_tile_index;
layout(location = 1) in vec2 v_tex_coord;
layout(location = 0) out vec4 f_color;
layout(set = 1, binding = 0) uniform texture2D t_tiles[];
layout(set = 1, binding = 1) uniform sampler s_tiles;

void main() {
    f_color = texture(sampler2D(t_tiles[nonuniformEXT(v_tile_index)], s_tiles), v_tex_coord);
}
````

And defining the following features:
```rust
features: wgpu::Features::
      wgpu::Features::SAMPLED_TEXTURE_BINDING_ARRAY
    | wgpu::Features::SAMPLED_TEXTURE_ARRAY_DYNAMIC_INDEXING
    | wgpu::Features::SAMPLED_TEXTURE_ARRAY_NON_UNIFORM_INDEXING
    | wgpu::Features::UNSIZED_BINDING_ARRAY,
```

I get the following panic when using the DX12 backend:
```
thread '<unnamed>' panicked at 'unsupported features were requested: SAMPLED_TEXTURE_ARRAY_DYNAMIC_INDEXING',
C:\Users\XXXXXX\.cargo\git\checkouts\wgpu-rs-40ea39809c03c5d8\0609e95\src\backend\direct.rs:1525:35
```

I can fix the panic under DX12 when using only the following features:
```rust
features: wgpu::Features::
      wgpu::Features::SAMPLED_TEXTURE_BINDING_ARRAY
    | wgpu::Features::SAMPLED_TEXTURE_ARRAY_NON_UNIFORM_INDEXING
    | wgpu::Features::UNSIZED_BINDING_ARRAY,
```

Which seems odd, since using `SAMPLED_TEXTURE_ARRAY_NON_UNIFORM_INDEXING` shouldn't exclude `SAMPLED_TEXTURE_ARRAY_DYNAMIC_INDEXING`, or at least this behavior seems not be documented.

**Comment by kvark** [comment]
Indeed! This looks like a problem on our side. cc @cwfitzgerald

**Comment by hasenbanck** [comment]
The issue has ben resolved using the new gfx-backend-dx12 v0.6.4.
